azbus/receiver.go

```diff
@@ -164,6 +218,62 @@ func (r *Receiver) String() string {
 	return fmt.Sprintf("%s", r.Cfg.TopicOrQueueName)
 }
 
+func (r *Receiver) receiveMessageBatches() error {
+	r.log.Debugf("BatchSize %d, BatchDeadline: %v", r.Cfg.BatchSize, r.Cfg.BatchDeadline)
+
+	for {
+		err := r.receiveOneMessageBatch()
+		if err != nil {
+			return err
+		}
+	}
+}
+
+func (r *Receiver) receiveOneMessageBatch() error {
+
+	if r.Cfg.BatchSize == 0 {
+		return fmt.Errorf("BatchSize must be greater than zero")
+	}
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
don't we have a proper context here?

Reply:
I think this is the equivalent of the other receiver but I will check